### ISSUE TITLE
Vertical scroll when swiping horizontally on daily/hourly charts

### ISSUE THREAD

**Issue by umk0m1qk** [breezy-weather/breezy-weather#1098]
### Steps to reproduce

Horizontal swipe navigation through daily and hourly tiles

### Expected behavior

Controlled horizontal and vertical swipe navigation that does not conflict

### Actual behavior

Very difficult to horizontal swipe through daily and hourly tiles without triggering vertical swipe. Suggest reducing sensitivity of vertical swipe action to avoid interference. 

### Weather source used

AccuWeather

### Breezy Weather version

5.2.4

### Android version

GrapheneOS build 2024051600

### Device

Pixel 8a

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue.
- [X] I checked that my issue is not covered in the [Frequently Asked Questions](https://github.com/breezy-weather/breezy-weather/blob/main/HELP.md) document.
- [X] I have updated the app to **[version 5.2.4 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)** and I can reproduce the issue in this version.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance), unless my issue only happens with a specific language. I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
Horizontal swipe on daily and hourly charts don't trigger a vertical scroll, on my side, even when swiping slightly in diagonal.

Please show a recording with visible touch points.

**Comment by elydpg** [comment]
I'm not the OP, but maybe they are referring to [this behaviour](https://github.com/breezy-weather/breezy-weather/assets/8622264/82de22b2-f922-498d-b4c4-6836ccd7b25e), where swiping even a tiny bit diagonally triggers an attempt to swipe to the next location screen

**Comment by papjul** [comment]
> I'm not the OP, but maybe they are referring to [this behaviour](https://github.com/breezy-weather/breezy-weather/assets/8622264/82de22b2-f922-498d-b4c4-6836ccd7b25e), where swiping even a tiny bit diagonally triggers an attempt to swipe to the next location screen

That's not the same thing, OP refers to a vertical swipe being triggered. The bouncing effect is normal, it's to prevent accidental swipes to next location, and it's working as intended

**Comment by H3Cki** [comment]
For me the collision with location swipe and card content swipe is very apparent annoying. If the swipable card content intercepted the swipe event immediately and cancelled the location swiping behavior completly there would be no problem imo.

**Comment by umk0m1qk** [comment]
> Horizontal swipe on daily and hourly charts don't trigger a vertical scroll, on my side, even when swiping slightly in diagonal.
> 
> Please show a recording with visible touch points.


https://github.com/breezy-weather/breezy-weather/assets/158820607/b1ee0943-c0e2-439e-8ba2-f628ef00b1cd

Maybe it's sloppy swiping on my part, but I don't struggle with other apps with similar layouts. Maybe the horizontal swipe zone is a little too narrow?

**Comment by papjul** [comment]
O_O

Is it something that happens in other apps?

I see that you have navigation gestures enabled, how does it behave if you temporarily switch to 3-buttons?

**Comment by umk0m1qk** [comment]
Switched to 3 button. I don't see a difference. That said, the more I play with it, the more my thumb & brain are calibrating to that narrow swipe band. Not having nearly as many unintended vertical scrolls as when I first started using the app. On day 1 of use, I felt like every other scroll attempt was a miss, and I got a bit frustrated. Probably just me.

As far as other apps, I don't recall encountering this issue.

**Comment by papjul** [comment]
I'm leaving this open to contributions if someone has an idea. Feel free to add any detail that might be worth for the investigation.

**Comment by rszyma** [comment]
A new user here, and noticed the same issue. My other point of comparison is Xiaomi stock weather app and the only difference between the two in terms of scrolling, is that in Xiaomi app, the vertical scroll roll has much lower resistance e.g. even lightest very light swipe goes on for about 2-3 seconds until stop, compared to about 0.5-1 second in breezy-weather.
Horizontal scroll however seems the same in both apps.

**Comment by papjul** [comment]
> Vertical scroll however seems the same in both apps.

You mean that when you swipe horizontally in a horizontal swipeable area on Xiaomi Weather app, it scrolls vertically as well? Is that correct?

**Comment by rszyma** [comment]
> > Vertical scroll however seems the same in both apps.

> You mean that when you swipe horizontally in a horizontal swipeable area on Xiaomi Weather app, it scrolls vertically as well? Is that correct?

No, I was just comparing scroll resistances between apps. There's no such issue actually that literally scrolling straight horizontally causes vertical scroll. We're talking about the same issue right? It's clearly shown on the video sent 6 posts above (https://github.com/breezy-weather/breezy-weather/issues/1098#issuecomment-2156705398). 

From my observation, generally, on Android horizontal scroll while in motion, allows gestures with some vertical-ish movement (let's say up to about 40-45°) to be counted towards further horizontal scroll; but as soon as the movement stops, trying to scroll horizontally with a the same, little bit of vertical movement will no longer result in horizontal, but vertical scroll. Hopefully that makes sense.

tldr: decreasing app-wide horizontal scroll resistance will probably solve this issue.

**Comment by rszyma** [comment]
> Maybe the horizontal swipe zone is a little too narrow?

@papjul is it possible to increase the swipe band? It's another thing that could help solve this issue

See the video below - should these swipes should count as vertical or horizontal?

https://github.com/breezy-weather/breezy-weather/assets/44318430/b26e2df5-8f59-4b3c-9050-ba01d665b6c3

**Comment by papjul** [comment]
> There's no such issue actually that literally swiping straight horizontally causes vertical scroll. We're talking about the same issue right? It's more about swiping diagonally. It's clearly shown on the video sent 6 posts above (https://github.com/breezy-weather/breezy-weather/issues/1098#issuecomment-2156705398).

No and yes. Swiping diagonally should not trigger a vertical scroll, as you described yourself:

> swipes with some vertical-ish movement (let's say up to about 40-45°) to be counted towards further horizontal scroll

It can be illustrated this way:
![image](https://github.com/breezy-weather/breezy-weather/assets/1449159/8ec4733e-3250-4567-8314-caa99bf8e8fb)

1. When swiping anywhere from area 1 to area 3 or 3 to 1, you scroll horizontally, but not vertically, not even slightly.
2. When swiping anywhere from area 2 to area 4 or 4 to 2, you scroll vertically, but not horizontally, not even slightly.

The video shows that there is a slightly diagonal swipe (I wouldn't say it's "clearly shown", it's very subtle IMO), but not one that would fall in the 2nd case, it's just about a 15° angle, far from the 45° angle.

But now that I try to reproduce again, I sometimes notice the same issue. But I have no idea what triggers it sometimes, and sometimes not.


> tldr: decreasing app-wide horizontal scroll resistance will probably solve this issue.

Can you give more technical details? I was pretty sure we just used the standard swiping system from Android SDK.

**Comment by papjul** [comment]
Fixed since v6.0.1-alpha

**Comment by papjul** [comment]
> If you try and be honest, you'll see that this is impossible to use as it is right now.

I struggled to produce the exact conditions to reproduce the issue, so yes, I assumed that being aware of the fix, you tried every possible way to trigger the issue, because the issue doesn't happen when I use the app normally.

> Geometric Weather did not have that issue.

Now, that changes a lot.

An analysis was done in #168, and it was wrongly assumed to be also present in Geometric Weather, which explained why it took so long to fix. But having tested Breezy Weather 4.0.5 and Breezy Weather 4.1.0, I can confirm the regression happened during the Kotlin rewrite. So, it should be easy to apply a fix that restores the old behavior. Reopening to investigate that.

> at least it doesnt happen anymore with only 1 location for me with the beta. I always wondered where it wanted to switch too with no additional locations 😀

I think you are talking about a different issue. We never had any horizontal swipe to next location with only 1 location. Please record a video so I understand if it needs a different issue to be opened.
